src/pages/dataElements/List.spec.tsx

```diff
@@ -0,0 +1,308 @@
+                            total: 1061,
+                            prevPage:
+                                '/dataElements/gist?pageListName=result&total=true&order=name:ASC&fields=access,id,name,code,domainType,valueType,lastUpdated,sharing&pageSize=20&page=1',
+                            nextPage:
+                                '/dataElements/gist?pageListName=result&total=true&order=name:ASC&fields=access,id,name,code,domainType,valueType,lastUpdated,sharing&pageSize=20&page=3',
+                            pageCount: 54,
+                        },
+                        result: [
+                            {
+                                ...filteredDataElementsMock.result[0],
+                                name: 'second page result',
+                            },
+                        ],
+                    }
+
+                    if (r.params.page === 1) {
+                        return Promise.resolve(firstPage)
+                    }
+                    if (r.params.page === 2) {
+                        return Promise.resolve(secondPage)
+                    }
+
+                    return Promise.reject('something wrong with mock')
+                },
+            }
+            return renderSection(dataProvider)
+        }
+        it('should display the page number', async () => {
+            const { getByTestId } = await renderWithPager()
+
+            const paginationFooter = getByTestId('section-list-pagination')
+
+            expect(paginationFooter).toHaveTextContent('Page 1 of 54')
+            expect(paginationFooter).toHaveTextContent('items 1-20 of 1061')
+        })
+        it('should disable previous page when on first page', async () => {
+            const { getByTestId } = await renderWithPager()
+            expect(
+                getByTestId('dhis2-uiwidgets-pagination-page-previous')
+            ).toHaveAttribute('disabled')
+        })
+        // next page
+        it('should allowing going to Next page', async () => {
+            const { getByTestId } = await renderWithPager()
+
+            userEvent.click(getByTestId('dhis2-uiwidgets-pagination-page-next'))
+
+            expect(pagerMock.mock.lastCall[0].params).toMatchObject(
+                expect.objectContaining({
+                    page: 2,
+                    pageListName: 'result',
+                    pageSize: 20,
+                    total: true,
+                })
+            )
+        })
+
+        // previous page
+        it('should allow going to Previous page ', async () => {
+            const { getByTestId, findByText, queryByText } =
+                await renderWithPager()
+
+            // first page loaded
+            await findByText('first page result')
+
+            // go to next page
+            userEvent.click(getByTestId('dhis2-uiwidgets-pagination-page-next'))
+            await findByText('second page result')
+            expect(queryByText('first page result')).toBeNull()
```

Comment:
This is a valid approach to get to the second page, although a bit more e2e-ish than integration-ish.
An alternative way of doing this would be to supply a custom history to react-router with the params in the url ([example](https://github.com/pbeshai/use-query-params/issues/80#issuecomment-615262267)).
That would also test whether a url is sharable and if url parameters are respected. When confirming at the end of the test that the url parameters have been adjusted, this simple test would cover a lot of aspects that are currently not being tested by this spec file